### PR TITLE
[Attributes] Add FloatAttributeType Feature

### PR DESCRIPTION
**Pull request by panigrc**
| Q               | A
| --------------- | -----
| Branch?         | master
| Bug fix?        | no
| New feature?    | yes
| BC breaks?      | no
| Deprecations?   | no
| License         | MIT

Selection_001

This feature adds a float attribute type. As future development it would be nice the [scale](https://symfony.com/doc/current/reference/forms/types/number.html#scale) option of  NumberType to be configurable.

### CODE REVIEW

**Review by pamil — COMMENTED** [review]
It looks good, however, we would need some tests for it, you can base them on our existing Behat scenarios:

```
features/product/managing_product_attributes/adding_integer_product_attribute.feature
features/product/managing_products/adding_product_with_integer_attribute.feature
```

Those are for `integer` attribute type, should be easy to create similar for `float` ones.

**Comment by panigrc** [comment]
@pamil thank you :slightly_smiling_face:. I will make the tests and will come back to you.

**Comment by Zales0123** [comment]
@panigrc thank you for the scenarios, but it would be great to also have implementation for them :) 🖖 ☮️

**Comment by panigrc** [comment]
@Zales0123 thank you for pointing that out, I saw that the Behat tests were failing.
I did a small change with the 1f39662a74880f0577f26ebf191128c7425b4c63 but I didn't find any other places in which I need to implement the scenarios.

I could use a little help to understand how to implement the scenarios.

**Comment by vvasiloi** [comment]
Hey @panigrc, do you still want to finish this? I can help with the tests.

**Comment by panigrc** [comment]
Hi @vvasiloi, sorry for not answering sooner your message, I forgot about this PR.
I haven't used Sylius for a while now and I think it's better to reject this PR.

**Comment by vvasiloi** [comment]
@panigrc no problem, maybe someone else will finish it.

**Comment by TheMilek** [comment]
Hi @panigrc!
Thank you for the PR 🍻 
I like the feature, I'm going to try to finish it 🚀

**Comment by panigrc** [comment]
@TheMilek  I'm grateful ! It's a long time ago since I pushed this PR.

File: features/product/managing_product_attributes/adding_float_product_attribute.feature [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    I want to add a new float product attribute
+
+    Background:
+        Given the store is available in "English (United States)"
+        And I am logged in as an administrator
+
+    @ui
+    Scenario: Adding a new float product attribute
+        Given I want to create a new float product attribute
+        When I specify its code as "display_size"
```
Comment (GSadee) [inline features/product/managing_product_attributes/adding_float_product_attribute.feature]:
```suggestion
        When I want to create a new float product attribute
        Given I specify its code as "display_size"
```

File: features/product/managing_products/adding_product_with_float_attribute.feature [hunk 2]
```diff
@@ -0,0 +1,22 @@
+@managing_products
+Feature: Adding a new product with an float attribute
```
Comment (GSadee) [inline features/product/managing_products/adding_product_with_float_attribute.feature]:
```suggestion
Feature: Adding a new product with a float attribute
```

File: features/product/managing_products/adding_product_with_float_attribute.feature [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    In order to extend my merchandise with more complex products
+    As an Administrator
+    I want to add a new product with an float attribute to the shop
```
Comment (GSadee) [inline features/product/managing_products/adding_product_with_float_attribute.feature]:
```suggestion
    I want to add a new product with a float attribute to the shop
```

File: features/product/managing_products/adding_product_with_float_attribute.feature [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+    Background:
+        Given the store operates on a single channel in "United States"
+        And the store has an float product attribute "Display Size"
+        And I am logged in as an administrator
+
+    @ui @javascript
+    Scenario: Adding an float attribute to product
```
Comment (GSadee) [inline features/product/managing_products/adding_product_with_float_attribute.feature]:
```suggestion
    Scenario: Adding a float attribute to a product
```

File: features/product/managing_products/adding_product_with_float_attribute.feature [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+        Given I want to create a new simple product
+        When I specify its code as "display_size"
```
Comment (GSadee) [inline features/product/managing_products/adding_product_with_float_attribute.feature]:
```suggestion
        When I want to create a new simple product
        Given I specify its code as "smartphone"
```

File: features/product/managing_products/adding_product_with_float_attribute.feature [hunk 6]
```diff
@@ -0,0 +1,22 @@
+@managing_products
+Feature: Adding a new product with an float attribute
+    In order to extend my merchandise with more complex products
+    As an Administrator
+    I want to add a new product with an float attribute to the shop
+
+    Background:
+        Given the store operates on a single channel in "United States"
+        And the store has an float product attribute "Display Size"
```
Comment (GSadee) [inline features/product/managing_products/adding_product_with_float_attribute.feature]:
```suggestion
        And the store has a float product attribute "Display Size"
```

**Review by GSadee — APPROVED** [review]

**Review by GSadee — APPROVED** [review]

**Comment by GSadee** [comment]
Thanks, Nik! :1st_place_medal:

**Pull request merged**